Hooks 🎉

CarsonF opened the pull request:
### `useSplitTreatment`
```tsx
const [treatment, config] = useSplitTreatment('my-split', { optionalAttributes });
return treatment === 'on' ? <MyFeature /> : null;
```
I didn't see a need to include multiple splits because the hook can just be called multiple times, which is inline with what React suggests.
~Maybe it would be good to return it as an array/tuple instead that way the individual treatments can be relabeled easier?~ Changed to that.

### `useSplitTrack`
```tsx
// Same track function as the client has
const track = useSplitTrack();
const handleClick = () => {
  track('button_clicked', ...);
};
```
Maybe `useTrack` would be better? And if it's ambiguous people can rename on import? I guess same goes for the other hook too.

CarsonF committed:
Created useSplit hook

CarsonF committed:
Created useTrack hook

CarsonF commented:
This kinda depends on #2, but I know that one has breaking changes. If we want this to go in first I should change the `client` check.

samuelcastro reviewed (approved):
It looks great, you mentioned breaking changes, do we have any breaking changes here?

CarsonF commented:
None here 😄

samuelcastro commented:
Another thing, we need to update the `README` to include the hooks usage, could you do that?

CarsonF committed:
Convert Split to use hooks

CarsonF commented:
Added docs and renamed the hooks.

We might still have a discrepancy with what's returned by default from `useSplit` and `Split`, the `"control"` vs `null` thing.

samuelcastro commented on README.md:
````diff
@@ -78,10 +78,27 @@ Split allows you to [implement a custom impression listener](https://help.split.
+const [feature1, config] = useSplit('feature1', { paying_customer: true });
+```
+
+### HOC
````
It's not an [HOC](https://reactjs.org/docs/higher-order-components.html) it's a [render props](https://reactjs.org/docs/render-props.html). I'd just add `### Component` people might not know the difference between HOC vs render props.

CarsonF replied:
My mistake. Will do!

samuelcastro reviewed (approved):
Great!
